collector/src/bin/collector.rs

```diff
@@ -663,6 +667,7 @@ fn main_result() -> anyhow::Result<i32> {
                     None,
```

Comment:
I think that we have gotten past the point where N `None`s in a row is an acceptable API :laughing: I know that it's mostly preexisting, but would you be willing to turn this e.g. into a builder pattern? Something like
```rust
get_local_toolchain(..., ToolchainConfig::default()) // everything `None` by default
get_local_toolchain(..., ToolchainConfig::default().rustc(...).clippy(...)) // some things filled
```
it could contain all parameters for the `get_local_toolchain` function, where it makes sense.